dvc/command/add.py

```diff
@@ -14,38 +15,40 @@ class CmdAdd(CmdBase):
     def run(self):
         try:
             if len(self.args.targets) > 1 and self.args.file:
-                raise DvcException("can't use '--file' with multiple targets")
-
-            for target in self.args.targets:
-                self.repo.add(
-                    target,
-                    recursive=self.args.recursive,
-                    no_commit=self.args.no_commit,
-                    fname=self.args.file,
+                raise RecursiveAddingWhileUsingFilename(
+                    "can't use '--file' with multiple targets"
                 )
 
-        except DvcException:
-            logger.exception("failed to add file")
+            self.repo.add(
+                self.args.targets,
+                recursive=self.args.recursive,
+                no_commit=self.args.no_commit,
+                fname=self.args.file,
+                progress=True,
+            )
+
+        except DvcException as err:
+            logger.exception("{}:{}".format(type(err).__name__, err))
```

Comment:
Why do you have to `format` exception here?

Reply:
no particular reason. would you prefer string addition or `':'.join` instead?

Reply:
@casperdcl Not sure I follow. `logger.exception` will automatically print last exception and format it with `:`, why do we need to format this separately?

Reply:
ah right. so just `logger.exception(err)`?

Reply:
@casperdcl if you only need to print the exception, then `logger.exception()`, err will be parsed automatically.